src/Tests/Microsoft.NET.TestFramework/ProjectConstruction/TestSolution.cs

```diff
@@ -0,0 +1,112 @@
+        {
+            Name = name;
+            SolutionPath = Path.Combine(newSolutionPath, $"{ name + ".sln"}");
+            Projects = new List<TestAsset>();
+            ProjectPaths = new List<string>();
+
+            var slnCreator = new DotnetNewCommand(log, "sln", "-o", $"{newSolutionPath}", "-n", $"{name}");
+            var slnCreationResult = slnCreator
+                .WithVirtualHive()
+                .Execute();
+
+            if(slnCreationResult.ExitCode != 0)
+            {
+                throw new Exception($"This test failed during a call to dotnet new. If {newSolutionPath} is valid, it's likely this test is failing because of dotnet new. If there are failing .NET new tests, please fix those and then see if this test still fails.");
+            }
+
+            foreach (var project in solutionProjects)
+            {
+                var slnProjectAdder = new DotnetCommand(log, "sln", $"{SolutionPath}", "add", Path.Combine(project.Path, project.TestProject.Name));
+                slnProjectAdder.Execute();
+                ProjectPaths.Add(project.Path);
+                Projects.Add(project);
+            }
+        }
+
+        /// <summary>
+        /// The FULL path to the newly created, on-disk solution, including the .sln file.
+        /// </summary>
+        public string SolutionPath { get; set; }
+
+        /// <summary>
+        /// The delegated or generated name of the solution.
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// A List of the paths of projects that were initially added to the solution.
+        /// </summary>
+        public List<string> ProjectPaths { get; set; }
+
+
+        /// <summary>
+        ///  The internal list of projects that contain actual testAssets.
+        ///  Not exposed publically to avoid incorrectly adding to this list or editing it without editing the solution file.
+        /// </summary>
+        internal List<TestAsset> Projects { get; set; }
+
+        /// <summary>
+        ///  Gives the property values for each project in the test solution.
+        ///  Can throw exceptions if the path to the project provided by tfm and configuration is not correct.
+        /// </summary>
+        /// <param name="targetFrameworksAndConfigurationsInOrderPerProject">A pair, first starting with the targetframework of the designated project.
+        /// The second item should be the Configuration expected of the second project.
+        /// This should match the order in which projects were added to the solution.
+        /// </param>
+        /// <returns>A dictionary of property -> value mappings for every subproject in the solution.</returns>
+        public List<Dictionary<string, string>> ProjectProperties(List<Tuple<string, string>> targetFrameworksAndConfigurationsInOrderPerProject = null)
```

Comment:
Suggest using tuples with named elements:

```suggestion
        public List<Dictionary<string, string>> ProjectProperties(List<(string targetFramework, string configuration)> targetFrameworksAndConfigurationsInOrderPerProject = null)
```